apps/projects/app/App.js

```diff
@@ -24,21 +24,22 @@ import {
 } from './store/eventTypes'
 
 import { initApolloClient } from './utils/apollo-client'
-import { getToken, getURLParam, githubPopup, STATUS } from './utils/github'
+import { getToken, githubPopup, STATUS } from './utils/github'
 import Unauthorized from './components/Content/Unauthorized'
 import { LoadingAnimation } from './components/Shared'
 import { EmptyWrapper } from './components/Shared'
 import { Error } from './components/Card'
 import { DecoratedReposProvider } from './context/DecoratedRepos'
 import usePathSegments from './hooks/usePathSegments'
 
+let popupRef = null
```

Comment:
Please, could you elaborate on why is it better to declare this outside the component and if you properly checked no potential issues could happen because putting this variable outside of the normal react lifecycles management.

What was wrong about having it in a React useState hook?

Reply:
When state changes, the component is re-rendered. We do not need that here. We only want to keep a reference to the pop-up so that we can verify incoming messages.

We could use `useRef`, but that is also more ceremony than we need. A global variable within this file is already encapsulated to this file, it is not accessible to other files because of how modules work. That’s perfect; that’s all we need.